src/query/storages/index/src/bloom.rs

```diff
@@ -53,193 +56,185 @@ use crate::SupportedType;
-        format!("Bloom({})", column_name)
-    }
-
-    pub fn build_filter_schema(data_schema: &DataSchema) -> DataSchema {
-        let mut filter_fields = vec![];
-        let fields = data_schema.fields();
-        for field in fields.iter() {
-            if Xor8Filter::is_supported_schema_type(field.data_type()) {
-                // create field for applicable ones
-
-                let column_name = Self::build_filter_column_name(field.name());
-                let filter_field = DataField::new(&column_name, SchemaDataType::String);
-
-                filter_fields.push(filter_field);
-            }
-        }
-
-        DataSchema::new(filter_fields)
-    }
-
     /// Load a filter directly from the source table's schema and the corresponding filter parquet file.
     #[tracing::instrument(level = "debug", skip_all)]
-    pub fn from_filter_chunk(
-        source_table_schema: DataSchemaRef,
-        filter_schema: DataSchemaRef,
-        filter_chunk: Chunk,
-    ) -> Result<Self> {
+    pub fn from_filter_chunk(fn_ctx: FunctionContext, filter_chunk: Chunk<String>) -> Result<Self> {
         Ok(Self {
-            source_schema: source_table_schema,
-            filter_schema,
             filter_chunk,
+            fn_ctx,
         })
     }
 
     /// Create a filter chunk from source data.
     ///
     /// All input chunks should belong to a Parquet file, e.g. the chunk array represents the parquet file in memory.
-    pub fn try_create(source_schema: DataSchemaRef, chunks: &[&Chunk]) -> Result<Self> {
+    pub fn try_create(fn_ctx: FunctionContext, chunks: &[&Chunk<String>]) -> Result<Self> {
         if chunks.is_empty() {
             return Err(ErrorCode::BadArguments("chunks is empty"));
         }
 
-        todo!("expression");
-    }
-
-    pub fn find(
-        &self,
-        column_name: &str,
-        target: &Scalar,
-        typ: &SchemaDataType,
-    ) -> Result<FilterEvalResult> {
-        let filter_column = Self::build_filter_column_name(column_name);
-        if !self.filter_schema.has_field(&filter_column)
-            || !Xor8Filter::is_supported_schema_type(typ)
-            || target.is_null()
-        {
-            // The column doesn't a filter
-            return Ok(FilterEvalResult::NotApplicable);
+        let mut filter_columns = vec![];
+
+        for i in 0..chunks[0].num_columns() {
+            if Xor8Filter::is_supported_type(&chunks[0].get_by_offset(i).data_type) {
+                // create filter per column
+                let mut filter_builder = Xor8Builder::create();
+
+                // ingest the same column data from all chunks
+                for chunk in chunks {
+                    let entry = chunk.get_by_offset(i);
+                    match &entry.value {
+                        Value::Scalar(scalar) => filter_builder.add_key(&scalar),
+                        Value::Column(col) => {
+                            col.iter()
```

Comment:
col.iter() will generate temp Enum ScalarRef which may lead to poor performance.

Maybe we could use `siphash` function to generate U64Column and add the u64 keys ?

It could be addressed after new expression is merged.

Reply:
> col.iter() will generate temp Enum ScalarRef which may lead to poor performance (hash of enum).

Does the performance penalty is caused by the `match` statement of the enum variants?
Are there any other factors related to the enum that can impact performance?

Reply:
#8196